From 80cff876fcc6253b7555b0ad723b14e90357c22d Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:13:47 -0400
Subject: [acpi] x86: Export 'boot_cpu_physical_apicid' to modules

Message-id: <1408594432-28006-13-git-send-email-ruwang@redhat.com>
Patchwork-id: 88489
O-Subject: [PATCH RHEL7.1 BZ1076655 12/17] x86: Export 'boot_cpu_physical_apicid' to modules
Bugzilla: 1076655
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

Backport of:
Original commit: cc08e04c3fdcaab767b8db27527002b2b4d758cb
Original Author: David Rientjes <rientjes@google.com>
Original Date: Thu, 14 Nov 2013 15:05:32 -0800

 Commit 9ebddac7ea2a "ACPI, x86: Fix extended error log driver to depend on
 CONFIG_X86_LOCAL_APIC" fixed a build error when CONFIG_X86_LOCAL_APIC was not
 selected and !CONFIG_SMP.

 However, since CONFIG_ACPI_EXTLOG is tristate, there is a second build error:

   ERROR: "boot_cpu_physical_apicid" [drivers/acpi/acpi_extlog.ko] undefined!

 The symbol needs to be exported for it to be available.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/apic/apic.c b/arch/x86/kernel/apic/apic.c
index db176bc..8adf12d 100644
--- a/arch/x86/kernel/apic/apic.c
+++ b/arch/x86/kernel/apic/apic.c
@@ -62,6 +62,7 @@ unsigned disabled_cpus;
 
 /* Processor that is doing the boot up */
 unsigned int boot_cpu_physical_apicid = -1U;
+EXPORT_SYMBOL_GPL(boot_cpu_physical_apicid);
 
 /*
  * The highest APIC ID seen during enumeration.
-- 
1.7.1